src/components/AutocompleteDropdown.tsx

```diff
@@ -0,0 +1,327 @@
+import * as React from 'react'
+import clsx from 'clsx'
+
+import useBooleanState from '../hooks/useBooleanState'
+
+type AutocompleteOption = {
+  label: string
+  value: string
+}
+
+type Props = {
+  id: string
+  label: string
+  value: unknown | undefined
+  placeholder: string | undefined
+  required: boolean | undefined
+  disabled: boolean | undefined
+  isLoading?: boolean
+  hasError?: boolean
+  validationMessage: string | undefined
+  displayValidationMessage: boolean
+  searchDebounceMs: number
+  searchMaxCharacters: number
+  onChangeValue: (newValue: string | undefined) => Promise<void> | void
+  onChangeLabel: (newLabel: string) => void
+  onSearch: (
+    label: string,
+    abortSignal: AbortSignal,
+  ) => Promise<AutocompleteOption[]>
+}
+
+function AutocompleteDropdown({
+  id,
+  label,
+  value,
+  placeholder,
+  required,
+  disabled,
+  validationMessage,
+  displayValidationMessage,
+  searchDebounceMs,
+  searchMaxCharacters,
```

Comment:
Not a big deal, but should this prop be called `searchMinCharacters` rather than `searchMaxCharacters`? Max kinda implies its the most characters you can search with?

Reply:
agreed